blog/_posts/2021-03-15-scaling-inlets.md

```diff
@@ -0,0 +1,349 @@
+For a couple of services, that's ok, but if we want to scale to tens or hundreds of tunnels, perhaps this can become an issue because of Let's Encrypt's rate limits.
+
+Because all our tunnels will be using the same domain, a wildcard certificate can help us here. If you have already such a certificate available, we can put that in place and otherwise, we can let cert-manager issue a wildcard certificate with the proper DNS01 Challenge Provider.
+
+## Pre-requisites
+
+- Some Kubernetes clusters running in different locations, e.g. on a public cloud (e.g. GKE, AKS, EKS, DOKS, …) or on a Raspberry Pi in a private home-lab
+- `kubectl`, configured to connect to the cluster
+- `kubectx`, optionally, but useful to manage the different cluster
+- `arkade` - portable Kubernetes marketplace
+- A domain and access to your DNS admin panel to create a sub-domain
+
+## Preparing the cluster
+
+Install the nginx ingress controller and cert-manager with `arkade`:
+
+``` bash
+arkade install ingress-nginx --namespace ingress-nginx
+arkade install cert-manager --namespace cert-manager
+```
+
+It can take some time until the nginx pods and loadbalancer is ready, but when everything is available, add the DNS records to connect your domain. Because we will create multiple tunnel for different applications, the easiest way to go forward is to configure a wildcard DNS record (e.g. *.inlets.example.com). If you don’t like wildcard records, create a record for each application pointing to the same public IP address (e.g. grafana.inlets.example.com, keycloak.inlets.example.com, …).
+
+Next, create an Issuer, or ClusterIssuer, to use Amazon Route53 to solve DNS01 ACME challenges.
+
+> A detailed explanation on how to prepare the issuer can be found [here](https://cert-manager.io/docs/configuration/acme/dns01/route53/)
+
+``` bash
+cat <<EOF | kubectl apply -f -
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: letsencrypt-prod
+spec:
+  acme:
+    server: https://acme-v02.api.letsencrypt.org/directory
+    email: <your email>
+    privateKeySecretRef:
+      name: letsencrypt-prod
+    solvers:
+    - dns01:
+        route53:
+          region: eu-central-1
+          accessKeyID: AKIAIOSFODNN7EXAMPLE
+          secretAccessKeySecretRef:
+            name: prod-route53-credentials-secret
+            key: secret-access-key
+EOF
+```
+
+Finally, create a wildcard Certificate resource with a reference the issuer created earlier:
+
+``` bash
+cat <<EOF | kubectl apply -f -
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: inlets-tls
+spec:
+  secretName: inlets-tls
+  issuerRef:
+    name: letsencrypt-prod
+    kind: Issuer
+  commonName: '*.inlets.example.com'
+  dnsNames:
+  - '*.inlets.example.com'
+EOF
+```
+
+Wait a few minutes until the cert-manager has done its job and the certificate is ready to use.
+
+``` bash
+$ kubectl describe certificate inlets-tls
```

Comment:
Please use a more concise: `kubectl get certificate inlets-tls` - we should focus on the broader task, this produces a lot of lengthy detail that's not required. `get cert` gives the Ready/Not ready info.